rust/datafusion/src/execution/physical_plan/parquet.rs

```diff
@@ -0,0 +1,254 @@
+                                            .send(Err(ExecutionError::General(format!(
+                                                "{:?}",
+                                                e
+                                            ))))
+                                            .unwrap();
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+
+                        Err(e) => {
+                            response_tx
+                                .send(Err(ExecutionError::General(format!("{:?}", e))))
+                                .unwrap();
+                        }
+                    }
+                }
+
+                Err(e) => {
+                    response_tx
+                        .send(Err(ExecutionError::General(format!("{:?}", e))))
+                        .unwrap();
+                }
+            }
+        });
+
+        let iterator = Arc::new(Mutex::new(ParquetIterator {
+            schema,
+            request_tx,
+            response_rx,
+        }));
+
+        Self { iterator }
+    }
+}
+
+impl Partition for ParquetPartition {
+    fn execute(&self) -> Result<Arc<Mutex<dyn BatchIterator>>> {
+        Ok(self.iterator.clone())
+    }
+}
+
+struct ParquetIterator {
+    schema: Arc<Schema>,
+    request_tx: Sender<()>,
+    response_rx: Receiver<Result<Option<RecordBatch>>>,
+}
+
+impl BatchIterator for ParquetIterator {
+    fn schema(&self) -> Arc<Schema> {
+        self.schema.clone()
+    }
+
+    fn next(&mut self) -> Result<Option<RecordBatch>> {
+        match self.request_tx.send(()) {
+            Ok(_) => match self.response_rx.recv() {
+                Ok(batch) => batch,
+                Err(e) => Err(ExecutionError::General(format!(
+                    "Error receiving batch: {:?}",
+                    e
+                ))),
+            },
+            _ => Err(ExecutionError::General(
+                "Error sending request for next batch".to_string(),
+            )),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::env;
+
+    #[test]
+    fn test() -> Result<()> {
+        let testdata =
+            env::var("PARQUET_TEST_DATA").expect("PARQUET_TEST_DATA not defined");
+        let filename = format!("{}/alltypes_plain.parquet", testdata);
+        let parquet_exec = ParquetExec::try_new(&filename, Some(vec![0, 1, 2]), 1024)?;
+        let partitions = parquet_exec.partitions()?;
+        assert_eq!(partitions.len(), 1);
+
+        let results = partitions[0].execute()?;
+        let mut results = results.lock().unwrap();
+        while let Some(batch) = results.next()? {
+            println!(
+                "got batch with {} rows and {} columns",
+                batch.num_rows(),
+                batch.num_columns()
+            );
+        }
+
+        //TODO assertions
```

Comment:
Are you going to add assertions in this PR?

Reply:
Yes. I'm away on a business trip today and tomorrow but intend on adding assertions this weekend.